src/remoted/shared_download.c

```diff
@@ -9,638 +9,479 @@
  * Foundation.
  */
 
-#include "shared.h"
 #include "shared_download.h"
-#include <pthread.h>
+#include "shared.h"
 
-OSHash *ptable;
-static remote_files_group *agent_remote_group;
-static agent_group *agents_group;
-static char yaml_file[OS_SIZE_1024 + 1];
-static time_t yaml_file_date;
-static ino_t yaml_file_inode;
-static pthread_mutex_t rem_yaml_mutex = PTHREAD_MUTEX_INITIALIZER;
+/** 
+ * @brief It deallocates config memory and make it points
+ * to config_tmp structure.
+ * 
+ * @param config The shared download configuration.
+ * @param config_tmp The temporal shared download configuration.
+ */
+static void sd_move(sd_config_t **config, sd_config_t **config_tmp);
 
-remote_files_group * w_parser_get_group(const char * name) {
-    remote_files_group * group = NULL;
+int sd_init(sd_config_t **config) {
 
-    w_mutex_lock(&rem_yaml_mutex);
+    os_calloc(1, sizeof(sd_config_t), *config);
+    (*config)->agents = NULL;
+    (*config)->groups = NULL;
+    (*config)->ptable = NULL;
 
-    if (ptable) {
-        group = OSHash_Get(ptable,name);
+    if ((*config)->ptable = OSHash_Create(), !(*config)->ptable) {
+        merror(W_PARSER_HASH_TABLE_ERROR);
+        return 0;
     }
 
-    w_mutex_unlock(&rem_yaml_mutex);
-    return group;
+    (*config)->mutex = (pthread_mutex_t) PTHREAD_MUTEX_INITIALIZER;
+
+    return 1;
 }
 
-agent_group * w_parser_get_agent(const char * name) {
-    agent_group * group = NULL;
+sd_group_t *sd_get_group(sd_config_t *config, const char *name) {
+    sd_group_t *group = NULL;
 
-    w_mutex_lock(&rem_yaml_mutex);
+    w_mutex_lock(&config->mutex);
```

Comment:
Some functions that manipulate config adquire the lock, whereas some others don't. We should go over this.